### ISSUE TITLE
Attribute Read Error

### ISSUE THREAD

**Issue by ChasenRanger** [LiorBanai/HDF5-CSharp#134]
While trying to read an attribute, I encountered a System.AccessViolationException: 'Attempted to read or write protected memory. This is often an indication that other memory is corrupt.'

The attribute in question was written using OpenCV, using the AtWrite() command to write a string, e.g. 'AtWrite("test", "test")'. OpenCV can successfully read the attribute back with "AtRead", as can h5py. I did test that this wasn't a one time error, and it happens each time I write the attribute, including if I switch it to "/test" rather than "test" as the location to write (and read). Note that AttributeExists() does return true on the attribute without an exception, its only trying to read it with "ReadStringAttributes" "ReadAttribute" or "ReadAttributes" that causes the exception.

**Comment by LiorBanai** [comment]
Can you attach example file for debugging?

**Comment by ChasenRanger** [comment]
I had to zip it to send it but here is an example .h5 file with a singular attribute called "test" with value "test" written using the openCV command AtWrite("test", "test"), which I verified does read back in h5py from the root directory.

[attestex.zip](https://github.com/LiorBanai/HDF5-CSharp/files/6703081/attestex.zip)

I've included the exact h5py code I used below:
```
>>> import h5py
>>> fil = h5py.File(r"{directorysnipped}\attestex.h5")
>>> fil.attrs
<Attributes of HDF5 object at 2016389422632>
>>> fil.attrs.keys()
<KeysViewHDF5 ['test']>
>>> fil.attrs['test']
b'test'
```

**Comment by LiorBanai** [comment]
Very strange. I don't understand for now what is the issue.
https://github.com/Apollo3zehn/HDF5.NET able to read the file so I'll check how it does so :)

**Comment by matthiasploner** [comment]
Hello I may have the same problem with the AccessViolationException, but it only occurs with attributes of fixed length. However, the Hdf5 viewer of the Hdf group displays the data and the API of Apollo3zehn can read the attributes. 
The Hdf5 file is generated by another software. 
I tried to write attributes with fixed length by changing the function WriteStringAttributes, but this did not work. Neither the Hdf5 viewer nor Apollo3zehn could read them.

Below I attach the HDF5 file and the modified line of the WriteStringAttributes function. I have renamed the file for uploading purpose. Just delete ".txt".

[input_2.h5.txt](https://github.com/LiorBanai/HDF5-CSharp/files/6830967/input_2.h5.txt)

```
//long datatype = H5T.create(H5T.class_t.STRING, H5T.VARIABLE);
long datatype = H5T.create(H5T.class_t.STRING, new IntPtr(values.ToArray()[0].Length));
```
